### ISSUE TITLE
[BUG] odd cases in Sum.maybe_none

### ISSUE THREAD

**Issue by sritchie** [genjax-community/genjax#1302]
A single entry returns a `Mask`:

```python
genjax.Sum.maybe_none(jnp.array(0), [None, None, None, 10])
# Mask(flag=Flag(f=<jax.Array(False, dtype=bool, weak_type=True)>), value=10)
```

No valid entry gives a `None`:

```python
genjax.Sum.maybe_none(jnp.array(0), [None, None])
# returns None
```

More than one option returns a `Sum`:

```python
genjax.Sum.maybe_none(jnp.array(0), [1,2,None])
# Sum(idx=<jax.Array(0, dtype=int32, weak_type=True)>, values=[1, 2, None])
```

All concrete options returns the thing:

```python
genjax.Sum.maybe_none(jnp.array(0), [1,2,3])
# Array(1, dtype=int32)
```

I suspect that the `Mask[R]` case is a bug, and that the `R` return case should go away, in favor of `Sum[R]` no matter what.
